src/main/java/com/synopsys/integration/detect/workflow/componentlocationanalysis/GenerateComponentLocationAnalysisOperation.java

```diff
@@ -58,7 +58,14 @@ public void locateComponentsForNonPersistentOnlineDetectorScan(List<DeveloperSca
      */
     public void locateComponentsForOfflineDetectorScan(BdioResult bdio, File scanOutputFolder, File projectSrcDir) throws ComponentLocatorException, DetectUserFriendlyException {
         List<Component> componentsList = bdioTransformer.transformBdioToComponentList(bdio);
-        runComponentLocator(componentsList, scanOutputFolder, projectSrcDir);
+        if(!componentsList.isEmpty())
```

Comment:
I recommend checking this condition before we get this far to avoid unnecessary work. What do you think about adding this conditional to this line instead? https://github.com/blackducksoftware/synopsys-detect/blob/d60a616d68539046915456b93382a29fed8fdd82/src/main/java/com/synopsys/integration/detect/lifecycle/run/operation/OperationRunner.java#L470

Reply:
This change is not needed. The parent method has the check:
generateComponentLocationAnalysisIfEnabled(...)

I am already working on this fix as part of IDETECT-3986.

Reply:
Yes! I did think about that but this is a use-case where bdio is in fact not empty but component list (dependency type in bdio is non existent) which seems to be the right thing to separate this use case and explicitly mention that with not the same message as here to the user.

Reply:
The parent method check if you debug does not yield 0, in fact it yields 1

Reply:
The very bug for IDETECT-3994 is that the check at the line I mentioned in my previous comment, which is the parent method Madhu is referring to, checks if there are code locations... and not whether or not any dependencies were found in any code location. Not sure what fix @gopannabd you are referring to exactly -- do you have a PR link I could look at?

@dmamidisynopsys Oh ok, yes I see... if we do wish to have a different msg "non-empty BDIO required" vs. "BDIO with OS components required". 

I was thinking we would instead reword the original msg so it encompasses both use cases? Since "empty-BDIO" and "void-of-dependencies-but-not-empty-BDIO" are the same in our eyes as far as this feature goes. But I am okay with having two different msgs, the functionality is the same, just more specific info msg. I'll leave that up to you. I would still recommend moving this conditional to "earlier" in the execution though to avoid unnecessary work by the transformer that creates `componentList`

Reply:
Specifically this - bdio.getCodeLocationNamesResult().getCodeLocationNames() yields 1

Reply:
We should provide clearly a different message. This is not empty bdio as suggested. I am good moving this to parent method, sure, but a different intuitive message should be presented.

Reply:
Just tested it out and we would have to _loop_ through each code location to confirm none of them have a OS dependency detected ... which contradicts my point above about avoiding "unnecessary work".  Approved